cluster-autoscaler/cloudprovider/gke/gke_manager.go

```diff
@@ -274,6 +274,8 @@ func (m *gkeManagerImpl) refreshNodePools(nodePools []NodePool) error {
 	existingMigs := map[gce.GceRef]struct{}{}
 	changed := false
 
+	m.cache.InvalidateTargetSizeCache()
```

Comment:
This is not the right place to invalidate. It _must_ be invalidated every loop and refreshNodePools is only called once in a while (https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/gke/gke_manager.go#L476).